pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -146,6 +146,14 @@ var (
 		Value: linux_defaults.RouteMarkDecrypt,
```

Comment:
As you mentioned offline, one risk of this cache approach is that someone may add a direct call to `netlink.XfrmState{Update,Add,Delete,Flush}` in the future, without going through the cache.

I looked at a couple way to prevent that in the code itself, but it seems complicated with Go. The best I could find would require us to import and redefine functions, methods, and constants from netlink in a new package. Then we could remove the import in `ipsec_linux.go` and replace it with a comment to never add it back :sweat_smile:  That all seems like a big pain for 4 functions.

Another approach would be to have some lint test for it. However I don't believe we have anything like that already. I'm unsure if it's worth it; it looks fairly unlikely someone would mess this up, but the consequences might also be too subtle to notice if it happens.

Reply:
> Another approach would be to have some lint test for it. However I don't believe we have anything like that already.

I think we have something like that, give me a sec.

Reply:
Added the test, it will definitely be useful for backports.